starknet_py/tests/e2e/client/full_node_test.py

```diff
@@ -400,3 +414,184 @@ async def test_get_syncing_status(full_node_client):
+    call = Call(
+        to_addr=deployed_balance_contract.address,
+        selector=get_selector_from_name("increase_balance"),
+        calldata=[0x10],
+    )
+    invoke_tx = await full_node_account.sign_invoke_transaction(
+        calls=call, auto_estimate=True
+    )
+    invoke_tx = dataclasses.replace(invoke_tx, signature=[])
+
+    simulated_txs = await full_node_account.client.simulate_transactions(
+        transactions=[invoke_tx], skip_validate=True, block_number="latest"
+    )
+    assert simulated_txs[0].transaction_trace.validate_invocation is None
+
+    with pytest.raises(ClientError, match=r".*INVALID_SIGNATURE_LENGTH.*"):
+        _ = await full_node_account.client.simulate_transactions(
+            transactions=[invoke_tx], block_number="latest"
+        )
+
+
+@pytest.mark.asyncio
+async def test_simulate_transactions_skip_fee_charge(
+    full_node_account, deployed_balance_contract
+):
+    assert isinstance(deployed_balance_contract, Contract)
+    call = Call(
+        to_addr=deployed_balance_contract.address,
+        selector=get_selector_from_name("increase_balance"),
+        calldata=[0x10],
+    )
+    invoke_tx = await full_node_account.sign_invoke_transaction(
+        calls=call, auto_estimate=True
+    )
+
+    # TODO (#1179): change this test
+    # because of python devnet, the SKIP_FEE_CHARGE flag isn't accepted
+    with pytest.raises(ClientError, match=r".*SKIP_FEE_CHARGE.*"):
+        _ = await full_node_account.client.simulate_transactions(
+            transactions=[invoke_tx], skip_fee_charge=True, block_number="latest"
+        )
+
+
+@pytest.mark.asyncio
+async def test_simulate_transactions_invoke(
+    full_node_account, deployed_balance_contract
+):
+    assert isinstance(deployed_balance_contract, Contract)
+    call = Call(
+        to_addr=deployed_balance_contract.address,
+        selector=get_selector_from_name("increase_balance"),
+        calldata=[0x10],
+    )
+    invoke_tx = await full_node_account.sign_invoke_transaction(
+        calls=call, auto_estimate=True
+    )
+    simulated_txs = await full_node_account.client.simulate_transactions(
+        transactions=[invoke_tx], block_number="latest"
+    )
+
+    assert isinstance(simulated_txs[0], SimulatedTransaction)
+    assert isinstance(simulated_txs[0].transaction_trace, InvokeTransactionTrace)
+    assert simulated_txs[0].fee_estimation.overall_fee > 0
+    assert simulated_txs[0].transaction_trace.validate_invocation is not None
+
+    invoke_tx = await full_node_account.sign_invoke_transaction(
+        calls=[call, call], auto_estimate=True
+    )
+    simulated_txs = await full_node_account.client.simulate_transactions(
+        transactions=[invoke_tx], block_number="latest"
+    )
+
+    assert isinstance(simulated_txs[0].transaction_trace, InvokeTransactionTrace)
+    assert simulated_txs[0].fee_estimation.overall_fee > 0
+    assert simulated_txs[0].transaction_trace.validate_invocation is not None
```

Comment:
Would it make sense to assert other fields of transaction_trace?

Reply:
it is a problem with devnet, because it doesn't return `fee_transfer_invocation` at all. I can add `execute_invocation` though